lib/compress/zstd_compress.c

```diff
@@ -2471,7 +2471,7 @@ static size_t ZSTD_compressBlock_targetCBlockSize(ZSTD_CCtx* zc,
      */
     if (cSize == 0) {
         cSize = ZSTD_noCompressSuperBlock(dst, dstCapacity, src, srcSize, zc->appliedParams.targetCBlockSize, lastBlock);
-        compressSuperBlock = 0;
+        bss = ZSTDbss_noCompress;
```

Comment:
@Cyan4973 I think maintaining a condition might be a reasonable option from the ones I considered for this after we talked. Creating a new bool wasn't necessary though. We can just reuse bss. The other options were:

1. Create a new function for the confirm repcodes and entropy block and invoke it within the branches as appropriate. I don't know if adding a new function call for a rather trivial operation is worth it. It doesn't necessarily make this much easier to read as the function call would have to be inside the branches (notably following the else at the bottom of the first branch). The upside of having a new function would mean we could call it from compressBlock_internal where the repocode and entropy checking code is currently duplicated. But this is an experimental branch so I don't know if changing something that part of the stable flow is a good idea for this feature's sake alone.

2. We could refactor both this function as well as frameChunk and compressBlock_internal to reduce code duplication. I think this would yield the most elegant solution but again, this is an experimental branch and changing parts of the stable flow might not be a good idea.